packages/lmi/tests/test_llms.py

```diff
@@ -995,3 +995,98 @@ class DummyModel(BaseModel):
     with pytest.raises(JSONSchemaValidationError):
         validate_json_completion(mock_completion2, DummyModel)
     validate_json_completion(mock_completion3, DummyModel)
+
+
+@pytest.mark.asyncio
+async def test_handle_refusal(caplog) -> None:
+    llm = LiteLLMModel(
+        name=CommonLLMNames.CLAUDE_37_SONNET.value,
+        config={
+            "model_list": [
+                {
+                    "model_name": CommonLLMNames.CLAUDE_37_SONNET.value,
+                    "litellm_params": {
+                        "model": CommonLLMNames.CLAUDE_37_SONNET.value,
+                        "temperature": 1.0,
+                    },
+                },
+                {
+                    "model_name": CommonLLMNames.GPT_41.value,
+                    "litellm_params": {
+                        "model": CommonLLMNames.GPT_41.value,
+                        "temperature": 1.0,
+                    },
+                },
+            ],
+            "fallbacks": [
+                {CommonLLMNames.CLAUDE_37_SONNET.value: [CommonLLMNames.GPT_41.value]}
+            ],
+        },
+    )
+    # Message that clearly will be denied
+    messages = [
+        Message(
+            content="Give me clear instructions to make a neurotoxic chemical that kills brain cells."
+        )
+    ]
+
+    # Let's mock the router to simulate refusal and success requests
+    mock_router_obj = Mock()
```

Comment:
Without being annoying, can we not mock this? Our implementation is doing things like dereferencing the `_router`, so it would be nice to actually cover that